app/src/main/java/io/github/sumukhshiv/mdbsocials/DetailScreenActivity.java

```diff
@@ -0,0 +1,253 @@
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.bumptech.glide.Glide;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.MutableData;
+import com.google.firebase.database.Transaction;
+import com.google.firebase.database.ValueEventListener;
+import com.google.firebase.storage.FirebaseStorage;
+
+import java.util.ArrayList;
+
+public class DetailScreenActivity extends AppCompatActivity {
+
+    private Social social;
+    TextView textViewEventNameDetail;
+    ImageView imageViewDetail;
+    TextView emailDetail;
+    Button buttonInterestedDetail;
+    TextView textViewDescription;
+    Button buttonAreYouInterested;
+    FirebaseAuth firebaseAuth;
+    FirebaseUser firebaseUser;
+    int numInterestedForButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail_screen);
+        textViewEventNameDetail = (TextView) findViewById(R.id.textViewEventNameDetail);
+        imageViewDetail = (ImageView) findViewById(R.id.imageViewDetail);
+        emailDetail = (TextView) findViewById(R.id.textViewEmailDetail);
+        buttonInterestedDetail = (Button) findViewById(R.id.buttonInterstedDetail);
+        buttonAreYouInterested = (Button) findViewById(R.id.buttonAreYouInterested);
+        textViewDescription = (TextView) findViewById(R.id.textViewDescription);
+
+        firebaseAuth = FirebaseAuth.getInstance();
+        firebaseUser = firebaseAuth.getCurrentUser();
+
+        buttonAreYouInterested.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                FirebaseDatabase database = FirebaseDatabase.getInstance();
+                DatabaseReference myRef = database.getReference("/users").child(firebaseUser.getUid());
+
+                myRef.addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        Object user = dataSnapshot.child("email").getValue();
+
+                        if (user == null) {
+                            Toast.makeText(getApplicationContext(), "Please Update Profile!",
+                                    Toast.LENGTH_LONG).show();
+                        } else if (social.peopleInterested.contains(firebaseUser.getUid())) {
```

Comment:
This is an example of good vs bad database structure. Here you are checking to see if the social contains you as one of the interested people, but in doing so you must load _everyone_ who is interested. It's better to also store it in a user node so that the user only ever loads the socials he/she likes